polar-core/src/resource_block.rs

```diff
@@ -167,15 +170,31 @@ pub fn resource_block_from_productions(
     }
 }
 
-#[derive(Clone, Debug)]
+#[derive(Clone, Debug, Serialize, Deserialize)]
 pub enum Declaration {
     Role,
     Permission,
     /// `Term` is a `Symbol` that is the (registered) type of the relation. E.g., `Org` in `parent: Org`.
     Relation(Term),
 }
 
-#[derive(Clone, Debug, Hash, PartialEq, Eq)]
+impl fmt::Display for Declaration {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            Self::Role => {
+                write!(f, "role")
+            }
+            Self::Permission => {
+                write!(f, "permission")
+            }
+            Self::Relation(_) => {
+                write!(f, "relation")
+            }
+        }
+    }
+}
```

Comment:
this should probably go in `formatting.rs` along with (many of) the existing `Display` impls

Reply:
moved into the `display` mod alongside the other `fmt::Display` impls